common/hash_data_other.go

```diff
@@ -0,0 +1,70 @@
+//go:build !windows
+// +build !windows
+
+package common
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func TryGetHashData(fullpath string) (SyncHashData, error) {
+	// get meta file name
+	dir, fn := filepath.Split(fullpath)
+	var metaFile string
+	if dir == "." { // Hide the file on UNIX-like systems (e.g. Linux, OSX)
+		metaFile = "./." + fn + AzCopyHashDataStream
+	} else {
+		metaFile = dir + "/." + fn + AzCopyHashDataStream
+	}
+	// open file for reading
+	f, err := os.OpenFile(metaFile, os.O_RDONLY, 0644)
+	if err != nil {
+		return SyncHashData{}, err
```

Comment:
In such cases where Get/Set of hash fails, what will be the impact on sync/copy operation. Do we fail the sync or assume hash is not available and overwrite the content?

Reply:
For initial hash generation, we let it fail. The user could be syncing from a read-only location.

Reply:
For putting the hash at the end of the transfer, we do not let it fail. The destination should be writeable anyway.